addons/hr/models/hr_employee_base.py

```diff
@@ -170,10 +170,8 @@ def _compute_presence_state(self):
 
     @api.depends('user_id')
     def _compute_last_activity(self):
-        presences = self.env['bus.presence'].search_read([('user_id', 'in', self.mapped('user_id').ids)], ['user_id', 'last_presence'])
-        # transform the result to a dict with this format {user.id: last_presence}
-        presences = {p['user_id'][0]: p['last_presence'] for p in presences}
-
+        # sudo: mail.presence - can access own user presence
+        presences = {p.user_id.id: p.last_presence for p in self.mapped("user_id").sudo().presence_id}
```

Comment:
"mapped" is not necessary to follow relational fields.
And creating a mapping is not necessary when it's strictly using the ORM and there is no manual query involved.
presence_id can directly be read on the target user in code below.